File: easy-problems/Python_solutions/monotonicArray.py
def getDir(arr):
	for i in range(len(arr) - 1):
		if arr[i] < arr[i + 1]:
			return 'inc'
		elif arr[i] > arr[i + 1]:
			return 'dec'


def isMonotonic(arr):
	initDir = getDir(arr)
	currDir = initDir

    for i in range(len(arr) - 1):
		if arr[i] < arr[i + 1]:
			currDir = 'inc'
		elif arr[i] > arr[i + 1]:
			currDir = 'dec'
		
		if currDir != initDir:
			return False
	return True

File: easy-problems/js-solutions/monotonicArray.js
const determineDir = (array) => {
  for (let i = 0; i < array.length - 1; i++) {
    if (array[i] < array[i + 1]) {
      return 'inc';
    } else if (array[i] > array[i + 1]) {
      return 'dec';
    }
  }
};

// iterate and determine direction of either increase or decreasing
function isMonotonic(array) {
  if (array.length <= 1) return true;
  let initialDir = determineDir(array);
  let currDir = initialDir;

  for (let i = 0; i < array.length - 1; i++) {
    if (array[i] < array[i + 1]) {
      currDir = 'inc';
    } else if (array[i] > array[i + 1]) {
      currDir = 'dec';
    }

    if (initialDir !== currDir) return false;
  }
  return true;
}
